tests/test_blueprint.py

```diff
@@ -652,6 +652,96 @@ def func():
             api.spec, 'response', 'DEFAULT_ERROR'
         )
 
+    @pytest.mark.parametrize('openapi_version', ['2.0', '3.0.2'])
+    def test_blueprint_error_schema(self, app, openapi_version, schemas):
+        """Check error schema is correctly documented"""
+        app.config['OPENAPI_VERSION'] = openapi_version
+        api = Api(app)
+        blp = Blueprint('test', 'test', url_prefix='/test')
+
+        example = {'error_id': 'E1', 'text': 'client error'}
+        examples = {
+            'example 1': {'error_id': 'E1', 'text': 'client error 1'},
+            'example 2': {'error_id': 'E2', 'text': 'client error 2'},
+        }
+
+        @blp.route('/')
+        @blp.error(400, schemas.ClientErrorSchema)
+        def func():
+            pass
+
+        @blp.route('/description')
+        @blp.error(400, schemas.ClientErrorSchema, description='Client error')
+        def func_with_description():
+            pass
+
+        @blp.route('/example')
+        @blp.error(400, schemas.ClientErrorSchema, example=example)
```

Comment:
can you add multiple @blp.errors?

Reply:
Yes absolutely. I ought to add a test to show it. Thanks for the feedback.

The decorator wraps the function, so if 10 error decorators are stacked, that means 10 nested function calls on each request. Not ideal but should be fast enough.

The wrapping is done to avoid modifying the function object itself which would yield wrong docs if people apply different decorators on the same view function. Not the main use case, but could happen.

If there is a perf penalty issue, we could create a decorator to document multiple errors at once. The syntax wouldn't be as nice but oh well. Let's not dive into unneeded optimization for now, anyway.

Reply:
👍

Reply:
Tested and fixed.